MediaBrowser.Controller/MediaEncoding/EncodingHelper.cs

```diff
@@ -1999,10 +2004,24 @@ public string GetMapArgs(EncodingJobInfo state)
 
             if (state.AudioStream != null)
             {
-                args += string.Format(
-                    CultureInfo.InvariantCulture,
-                    " -map 0:{0}",
-                    state.AudioStream.Index);
+                if (state.AudioStream.IsExternal)
+                {
+                    int externalAudioMapIndex = state.SubtitleStream != null && state.SubtitleStream.IsExternal ? 2 : 1;
+                    int externalAudioStream = state.MediaSource.MediaStreams.Where(i => i.Path == state.AudioStream.Path).ToList().IndexOf(state.AudioStream);
```

Comment:
```suggestion
                    int externalAudioStream = state.MediaSource.MediaStreams.FindIndex(i => i.Path == state.AudioStream.Path);
```
Shouldn't there be only one audio stream that has the same path?

Reply:
No, `.mka` files can have multiple audio streams in one file (didn't know that either until @cvium told me) :)